### PR TITLE
Add tensor buffer operations.

### PR DESCRIPTION
**Pull request by toregge**
@geirst : please review
@havardpe : FYI

### CODE REVIEW

File: searchlib/src/vespa/searchlib/tensor/tensor_buffer_operations.h [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+struct Value;
+class ValueType;
+}
+
+namespace search::tensor {
+
+/*
+ * Class used to store a tensor in a buffer and make tensor views based on
+ * buffer content.
+ */
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/tensor/tensor_buffer_operations.h:26]: Would be nice with a short explanation of the memory format here as well.

File: searchlib/src/vespa/searchlib/tensor/tensor_buffer_operations.cpp [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+namespace {
+
+uint32_t
+adjust_min_alignment(size_t min_alignment)
+{
+    // Also apply alignment for num_subspaces and labels
+    return std::max(std::max(sizeof(uint32_t), sizeof(string_id)), min_alignment);
+}
+
+struct MyFastValueView final : Value {
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/tensor/tensor_buffer_operations.cpp:38]: Would just _FastValueView_ be sufficient here?

**Review by geirst — APPROVED** [review]
👍

File: searchlib/src/vespa/searchlib/tensor/tensor_buffer_operations.h (hunk 1 above)
Comment (toregge) [reply searchlib/src/vespa/searchlib/tensor/tensor_buffer_operations.h:37]: Short description added.

File: searchlib/src/vespa/searchlib/tensor/tensor_buffer_operations.cpp (hunk 2 above)
Comment (toregge) [reply searchlib/src/vespa/searchlib/tensor/tensor_buffer_operations.cpp]: Renamed.

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**